src/uploader.c

```diff
@@ -1492,6 +1572,192 @@ static void queue_request_frame_id(storj_upload_state_t *state)
     state->requesting_frame = true;
 }
 
+static void after_create_parity_shards(uv_work_t *work, int status)
+{
+    parity_shard_req_t *req = work->data;
+    storj_upload_state_t *state = req->upload_state;
+
+    state->pending_work_count -= 1;
+    state->create_parity_shard_count += 1;
+
+    // TODO: Check if file was created
+    if (req->error_status != 0) {
+        state->log->warn(state->env->log_options, state->handle,
+                       "Something went wrong!!");
+
+        state->awaiting_parity_shards = true;
+
+        if (state->create_parity_shard_count == 6) {
+            state->error_status = STORJ_FILE_SIZE_ERROR;
+        }
+    } else {
+        state->log->info(state->env->log_options, state->handle,
+                       "Successfully created parity shards");
+
+        state->parity_file = fopen(state->parity_file_path, "r");
+    }
+
+clean_variables:
+    queue_next_work(state);
+    free(work);
+}
+
+static void create_parity_shards(uv_work_t *work)
+{
+    parity_shard_req_t *req = work->data;
+    storj_upload_state_t *state = req->upload_state;
+
+    state->log->info(state->env->log_options, state->handle,
+                   "Creating parity shards");
+
+    // ???
+    fec_init();
+
+    uint8_t *map = NULL;
+    int status = 0;
+    status = map_file(fileno(state->original_file), state->file_size, &map, true);
+
+    if (status) {
+        req->error_status = 1;
+        state->log->error(state->env->log_options, state->handle,
+                       "Could not create mmap original file: %d", status);
+        goto clean_variables;
+    }
+
+    uint64_t block_size = (state->file_size + state->total_data_shards - 1) / state->total_data_shards;
```

Comment:
After testing, the `block_size` here doesn't match `shard_size` as used to upload each shard.